packages/internals/src/engine-commands/getConfig.ts

```diff
@@ -142,3 +148,29 @@ export async function getConfig(options: GetConfigOptions): Promise<ConfigMetaFo
 
   throw error
 }
+
+async function resolveBinaryTargets(generator: GeneratorConfig) {
```

Comment:
Resolves `binaryTargets` eagerly instead of lazily, to avoid discrepancies between the `getConfig` value given by the generator and the raw `getConfig` value which causes differences if the actual generated code of the clients...

Reply:
Just to clarify:
- `binaryTargets` is resolved eagerly
- the `value` property of binary targets when they are native w.r.t. the current platform or when `binaryTargets` is empy, is resolved via the memoized helper `getPlatform()`

Reply:
> binaryTargets is resolved eagerly

Yes. And if a `binaryTarget` is an env var, it will try to resolve it eagerly if it can. Otherwise, will leave it unresolved (`value = null`, `fromEnvVar = something`).

> the value property of binary targets when they are native w.r.t. the current platform or when binaryTargets is empy, is resolved via the memoized helper getPlatform()

Yes, if we find `native` we directly transform it to the current platform. And if we have the default (nothing provided), we add the default for `native` (current platform).